### ISSUE TITLE
Dynamic models: Possible memory issues because of _subclasses

### ISSUE THREAD

**Issue by mjrk** [schematics/schematics#501]
In https://github.com/schematics/schematics/blob/v2.0.1/schematics/models.py#L118, we find this code

```python
        # Register class on ancestor models
        klass._subclasses = []
        for base in klass.__mro__[1:]:
            if isinstance(base, ModelMeta):
            base._subclasses.append(klass)
```

Now, I have the case where I need to dynamically create models out of some JSON for validation and transformation (i.e. using https://github.com/retresco/acspec). It worked like charm but the lines above cause a memory issue: every new definition of a Model will add a reference to itself to the parent class. This has the effect, that the _subclasses list get unlimitedly long and that the class definition is not garbage collected later on. Memory usage increases heavily. 

If I remove/comment out the lines above, the list won't get big and classes are garbage collected when not in use anymore. (There is still a small, constant increase in memory though that I would have to debug.)

The `_subclasses` list is only used for polymorhic models, correct?

It would be great if this referencing could be turned of in some way or maybe even an alternative implementation without this requirement could help out with dynamic models.

Could a PR with a new option `parent_subclass_reference=False/True` be accepted?

**Comment by lkraider** [comment]
Indeed, PolyModelType needs to be reworked, as mentioned [here](https://github.com/schematics/schematics/issues/448#issuecomment-258643850).

A smarter way to handle the subclasses would be welcome - or probably not store a copy, but iterate the mro in the PolyModelType method?

**Comment by mjrk** [comment]
New-style classes have the `__subclasses__` attribute. Maybe you don't need to add them your own but rely on this attribute only? Not sure if it works for you though and how it will combine with garbage collection.

**Comment by mjrk** [comment]
Here a PR to use the build-in `__subclasses__()` method: https://github.com/schematics/schematics/pull/502

This would basically fix the issue with dynamic models for the moment.

(Btw. any chance that this could be backported to 0.1.X versions?)

**Comment by lkraider** [comment]
We can target a v1.1.3 release with this fix. You can create a PR and target the v1.1.2 tag as base.

**Comment by mjrk** [comment]
Apparently I can choose a tag for compare, but cannot create a PR https://github.com/schematics/schematics/compare/v1.1.2...mjrk:feauture/use-__subclasses__1

Maybe we need a branch for this?

**Comment by lkraider** [comment]
Yeah seems so, I've created a `1.1.3` branch for this.

**Comment by mjrk** [comment]
Thanks so much! Here the PR https://github.com/schematics/schematics/pull/503

And in general a lot of thanks for your efforts here in schematics!

**Comment by mjrk** [comment]
Notes for the PR:

The travis build https://travis-ci.org/schematics/schematics/jobs/246509018 for https://github.com/schematics/schematics/pull/503 failed with `ERROR: InterpreterNotFound: python2.6`. I guess, python2.6 used to be preinstalled but has to be mentioned explicitly now in the `.travis` yaml python-section. So I added the matrix from the current schematics version to the .travis.yml.

However, Sphinx dropped compatibility with python2.6 at version 1.5, so I had to restricted the Sphinx version
